### PR TITLE
Implement the text format so that w3c trace context headers are propagated.

### PR DESCRIPTION
**Pull request by jkwatson**

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/distributedcontext/DistributedContextHttpTextFormat.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/** W3C Trace Context implementation of the HttpTextFormat for a DistributedContext. */
+class DistributedContextHttpTextFormat implements HttpTextFormat<DistributedContext> {
+  static final String TRACEPARENT = "traceparent";
+  static final String TRACESTATE = "tracestate";
+  private static final List<String> FIELDS =
+      Collections.unmodifiableList(Arrays.asList(TRACEPARENT, TRACESTATE));
```
Comment (bogdandrutu) [inline sdk/src/main/java/io/opentelemetry/sdk/distributedcontext/DistributedContextHttpTextFormat.java]: I think this is the correlation-context from the w3c.

Comment (jkwatson) [reply sdk/src/main/java/io/opentelemetry/sdk/distributedcontext/DistributedContextHttpTextFormat.java]: Is there something you think should be renamed in here?

**Comment by arminru** [comment]
Hi John!

W3C Trace Context (text format) is already implemented here:
https://github.com/open-telemetry/opentelemetry-java/blob/master/api/src/main/java/io/opentelemetry/trace/propagation/HttpTraceContext.java

The DistributedContext you're trying to implement is a compound of the W3C Correlation Context (currently being drafted [here](https://w3c.github.io/correlation-context)) on the one hand and what is commonly referred to as "Baggage" on the other hand, but not the W3C Trace Context (unless I'm mixing things up myself here).
There's a proposal to properly split DistributedContext into Correlation Context and Baggage being discussed here: https://github.com/open-telemetry/oteps/pull/42. It might be better to wait for this to be settled.

See also:
- https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/overview.md#distributedcontext
- https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-distributedcontext.md
- https://github.com/open-telemetry/oteps/pull/42
- https://github.com/open-telemetry/oteps/pull/36

**Comment by jkwatson** [comment]
Sorry, the title of the PR is probably a little misleading. The purpose here was to at least propagate the trace context headers, so traces aren't broken. With the current implementation (that just throws UnsupportedOperationExceptions), traces are broken when requests flow through the tracers. 

I'm happy to update any and all verbiage as appropriate, or simply have the methods return empty contexts and not inject anything (still resulting in broken traces, but at least not throwing exceptions).

**Comment by jkwatson** [comment]
I was confused about where the headers needed to be propagated. This is actually fixed for real in #608 . closing!

**Pull request closed without merging**